### PR TITLE
[Feature] Empty Programming Exercises

### LINKED ISSUE #3308: Programming Language -- None
### Is your feature request related to a problem?
I'm using Artemis' programming exercises for the exercises in Security Engineering.
It is way more convenient to provide the scripts, program stubs,  etc. via Artemis rather than via a ZIP archive on Moodle or setting up a new git repository on gitlab.lrz.de.

The problems are that ...

1. Automatic tests are somewhat difficult to create for the exercises in Security Engineering and
2. The programming languages used, e.g., PHP to demonstrate a SQL injection, are different from Artemis' ones.
Artemis languages

### Describe the solution you'd like

I would love to have a `None` or `empty` choice in the context menu.
Artemis would then generate an empty template, solution, and test repository.

### Describe alternatives you've considered

Currently, I choose `Python` for those exercises, and then  I delete the python files.
Thus, this is also not that much of an overhead.

### PR DESCRIPTION
**Pull request by ivanchimeno**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context

Resolves #3308 

### Description

This PR adds support to create empty programming exercises. It adds the "Empty" drop-down item in the programming language list and generates empty template, test, and solution repositories. It also adds a dummy problem statement and a bare-bone build plan for Bamboo and Jenkins

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. Click on Generate New Programming Exercise
4. Select "Empty" from the drop-down list of the programming language
5. Generate the exercise
6. Make sure that the template, tests, and solution repositories are created. They should also be empty.

_**Please delete the empty exercise after finishing testing it.**_

### Test Coverage

BambooBuildPlanService 95%
JenkinsBuildPlanService 99%
programming-exercise-update.component.ts 79.76%

### Screenshots

image

### CODE REVIEW

**Review by maximilianbergmann — DISMISSED** [review]
tested on ts3. works as expected

**Review by KiraNickel — APPROVED** [review]
Tested on ts3. Works as described 👍🏼

**Review by katjanakosic — APPROVED** [review]
Worked as expected on TS2

File: src/main/resources/templates/empty/readme [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Empty Template Problem Statement
+
+Insert problem statement here here.
```
Comment (FuchsDominik) [inline src/main/resources/templates/empty/readme:3]:
```suggestion
Insert problem statement here.
```

**Review by MDK5 — APPROVED** [review]
Reviewed code, changes look good to me

**Review by imstreetspirit — CHANGES_REQUESTED** [review]
code seems good in general. I noticed the following translation issue in the programming-exercise-detail page. 


Maybe its not relevant and out of scope for this task, but in the exercise detail page it says build failed for template and solution. Yeah, i know they are empty, maybe build can not be triggered if the repos are empty, dunno :)

**Review by nicolasruscher — COMMENTED** [review]
could not find any additional problems. I will approve once the translation issue is fixed.

**Review by SimonEntholzer — APPROVED** [review]
Looks good, worked on TS3. Accessing the repositories only worked by using the 'edit in editor' feature, the links to the repos on bitbucket don't work and the links on the right lead to the course overview page. Since this is the case for every exercsie type for some reason, I don't think it should be your concern in this PR.
image

**Review by Malyuk-A — APPROVED** [review]
Tested on TS1. Works as expected & the code looks good.

**Comment by ivanchimeno** [comment]
> Looks good, worked on TS3. Accessing the repositories only worked by using the 'edit in editor' feature, the links to the repos on bitbucket don't work and the links on the right lead to the course overview page. Since this is the case for every exercsie type for some reason, I don't think it should be your concern in this PR.
> image

It's the same behaviour for other programming languages. I think you need to refresh the page.

**Review by NicolasUllmann — DISMISSED** [review]
Tested on Server 1. Everything worked as expected.

**Review by canyelten — DISMISSED** [review]
Tested on TS1, works as expected.

**Review by willergeorg — DISMISSED** [review]
Tested on TS1. Works for me.

**Review by imstreetspirit — DISMISSED** [review]
changes look good to me

**Review by KiraNickel — DISMISSED** [review]
Tested on ts3, everything works as expected 👍🏼

**Review by ndricimrr — APPROVED** [review]
Tested on TS2 and works as expected

**Review by argertboja — APPROVED** [review]
Tested on TS2, reviewed the code and everything LGTM 👍🏼

**Review by katjanakosic — APPROVED** [review]
Worked as expected on TS1

**Review by MaisiKoleni — APPROVED** [review]
Code looks good to me. It is only a bit strange that we use the Java Docker image, but this will likely not matter to most users.

File: src/main/java/de/tum/in/www1/artemis/service/connectors/ContinuousIntegrationService.java [hunk 2]
```diff
@@ -287,7 +287,7 @@ public String forProgrammingLanguage(ProgrammingLanguage language) {
      */
     static String getDockerImageName(ProgrammingLanguage language) {
         return switch (language) {
-            case JAVA, KOTLIN -> "ls1tum/artemis-maven-template:java16-2";
+            case JAVA, KOTLIN, EMPTY -> "ls1tum/artemis-maven-template:java16-2";
```
[the remark this answers is not in the document]
Comment (ivanchimeno) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/ContinuousIntegrationService.java:290]: Yes I wanted to add a simple boilerplate build plan that instructors can extend

**Review by nicolasruscher — APPROVED** [review]
The code looks good to me

File: src/main/resources/templates/jenkins/empty/regularRuns/Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,38 @@
+// ARTEMIS: JenkinsPipeline
+
+pipeline {
+        options {
+            timeout(time: #jenkinsTimeout, unit: 'MINUTES')
+        }
+        agent {
+            docker {
+                image 'maven:3.3.3'
+                label 'docker'
```
Comment (jpbernius) [inline src/main/resources/templates/jenkins/empty/regularRuns/Jenkinsfile:10]: Shouldn't we use the same Docker Image as defined in `ContinuousIntegrationService.java`?

Comment (ivanchimeno) [reply src/main/resources/templates/jenkins/empty/regularRuns/Jenkinsfile:10]: We can do that yes

**Review by jpbernius — APPROVED** [review]

**Review by nicolasruscher — APPROVED** [review]

**Pull request merged**